commit 0f5f93bd33799a39b24a09d981a3b567b5f8581c
Author: Albert Astals Cid <aacid@kde.org>
Date:   Thu Jul 4 19:37:57 2019 +0200

    REVERT
    Remove ifdef for Qt 5.11 since that is now the required version

diff --git src/delegaterecycler.cpp src/delegaterecycler.cpp
index 0f59ccf..688c704 100644
--- src/delegaterecycler.cpp
+++ src/delegaterecycler.cpp
@@ -289,10 +289,16 @@ void DelegateRecycler::setSourceComponent(QQmlComponent *component)
         syncModel();
 
         QQmlContext *ctx = QQmlEngine::contextForObject(m_item)->parentContext();
+#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
+        ctx->setContextProperty(QStringLiteral("modelData"), m_propertiesTracker->property("trackedModelData"));
+        ctx->setContextProperty(QStringLiteral("index"), m_propertiesTracker->property("trackedIndex"));
+        ctx->setContextProperty(QStringLiteral("delegateRecycler"), QVariant::fromValue<QObject*>(this));
+#else
         ctx->setContextProperties({ QQmlContext::PropertyPair{ QStringLiteral("modelData"), m_propertiesTracker->property("trackedModelData") },
                                     QQmlContext::PropertyPair{ QStringLiteral("index"), m_propertiesTracker->property("trackedIndex")},
                                     QQmlContext::PropertyPair{ QStringLiteral("delegateRecycler"), QVariant::fromValue<QObject*>(this) }
                                  });
+#endif
     }
 
     if (m_item) {
diff --git src/columnview.cpp src/columnview.cpp
index d7ebce7..8221434 100644
--- src/columnview.cpp
+++ src/columnview.cpp
@@ -29,6 +29,19 @@
 #include <QDebug>
 #include <QPropertyAnimation>
 
+template <class QtType>
+class Q510QI : public QtType
+{
+public:
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
+    QSizeF size()
+    {
+        return QSizeF(QtType::width(), QtType::height());
+    }
+#endif
+};
+#define Q510QIPCAST(instanceptr) static_cast<Q510QI<std::decay<decltype(*instanceptr)>::type>* >(instanceptr)
+
 
 QHash<QObject *, ColumnViewAttached *> ColumnView::m_attachedObjects = QHash<QObject *, ColumnViewAttached *>();
 
@@ -283,8 +296,9 @@ ContentItem::ContentItem(ColumnView *parent)
         if (!m_view->currentItem()) {
             m_view->setCurrentIndex(m_items.indexOf(m_viewAnchorItem));
         } else {
-            QRectF mapped = m_view->currentItem()->mapRectToItem(m_view, QRectF(QPointF(0, 0), m_view->currentItem()->size()));
-            if (!QRectF(QPointF(0, 0), m_view->size()).intersects(mapped)) {
+            QRectF mapped = m_view->currentItem()->mapRectToItem(m_view, QRectF(QPointF(0, 0),
+                Q510QIPCAST(m_view->currentItem())->size()));
+            if (!QRectF(QPointF(0, 0), Q510QIPCAST(m_view)->size()).intersects(mapped)) {
                 m_view->setCurrentIndex(m_items.indexOf(m_viewAnchorItem));
             }
         }
@@ -781,7 +795,7 @@ void ColumnView::setCurrentIndex(int index)
         // If the current item is not on view, scroll
         QRectF mappedCurrent = m_currentItem->mapRectToItem(this,
                                             QRectF(QPointF(0, 0),
-                                                   m_currentItem->size()));
+                                                   static_cast<Q510QI<QQuickItem>* >(m_currentItem.data())->size()));
 
         if (m_contentItem->m_slideAnim->state() == QAbstractAnimation::Running) {
             mappedCurrent.moveLeft(mappedCurrent.left() + m_contentItem->x() + m_contentItem->m_slideAnim->endValue().toInt());
